scripts/analysis.py

```diff
@@ -250,229 +116,198 @@ def get_sim_time_duration(cursor):
-
-    return temp_isotope, temp_mass, time_list
-
-
-def plot_in_out_flux(cursor, facility, influx_bool, title, outputname):
-    """plots timeseries outflux from facility name in kg.
+def get_timeseries(in_list, duration, multiplyby, cumulative):
+    """ returns a timeseries list of a given data
 
     Parameters
     ----------
-    cursor: sqlite cursor
-        sqlite cursor
-    facility: str
-        facility name
-    influx_bool: bool
-        if true, calculates influx,
-        if false, calculates outflux
-    title: str
-        title of the multi line plot
-    outputname: str
-        filename of the multi line plot file
+    in_list: list
+        list of data to be created into timeseries
+        list[0] = time
+        list[1] = value, quantity
+    duration: int
+        duration of the simulation
+    multiplyby: int
+        integer to multiply the value in the list by for
+        unit conversion from kilograms
+    cumulative: boolean
+        determine whether returned timeseries should be
+        cumulative sum over time
 
     Returns
     -------
-
+    timeseries list of data
     """
-
-    cur = cursor
-    agent_ids = get_agent_ids(cur, facility)
-    if influx_bool is True:
-        resources = cur.execute(exec_string(agent_ids,
-                                            'transactions.receiverId',
-                                            'sum(quantity), time, qualid')
-                                + ' GROUP BY time, qualid').fetchall()
+    value = 0
+    value_timeseries = []
+    array = np.array(in_list)
+    if cumulative:
+        for i in range(0, duration):
+            if len(array) > 0:
+                value += sum(array[array[:, 0] == i][:, 1])
+            value_timeseries.append(value * multiplyby)
     else:
-        resources = cur.execute(exec_string(agent_ids,
-                                            'transactions.senderId',
-                                            'sum(quantity), time, qualid')
-                                + ' GROUP BY time, qualid').fetchall()
-
-    compositions = cur.execute('SELECT qualid, nucid, massfrac\
-                               FROM compositions').fetchall()
+        for i in range(0, duration):
+            value = sum(array[array[:, 0] == i][:, 1])
+            value_timeseries.append(value * multiplyby)
 
-    init_year, init_month, duration, timestep = get_sim_time_duration(cur)
-    isotope, mass, time_list = isotope_mass_time_list(resources, compositions)
+    return value_timeseries
 
-    waste_dict = get_waste_dict(isotope, mass, time_list, duration)
 
-    if influx_bool is False:
-        stacked_bar_chart(waste_dict, timestep,
-                          'Years', 'Mass [kg]',
-                          title, outputname, init_year)
-    else:
-        multi_line_plot(waste_dict, timestep,
-                        'Years', 'Mass [kg]',
-                        title, outputname, init_year)
+"""GETTERS"""
```

Comment:
I'm not sure comments are any better. They may end up just cruft. If the set of functions need to be separated into types, because the file is too long or because they serve different purposes, perhaps those groups of functions should be in separate files.